backend/de.metas.deliveryplanning.base/src/main/java/de/metas/deliveryplanning/DeliveryPlanningRepository.java

```diff
@@ -589,4 +593,32 @@ public boolean hasCompleteDeliveryInstruction(@NonNull final DeliveryPlanningId
 				.addEqualsFilter(I_M_ShipperTransportation.COLUMNNAME_DocStatus, DocStatus.Completed)
 				.anyMatch();
 	}
+
+	public void updateICFromDeliveryPlanningId(@NonNull final DeliveryPlanningId deliveryPlanningId)
+	{
+		final I_M_Delivery_Planning currentDeliveryPlanning = getById(deliveryPlanningId);
+		final OrderLineId orderLineId = OrderLineId.ofRepoIdOrNull(currentDeliveryPlanning.getC_OrderLine_ID());
+		if (orderLineId == null)
+		{
+			return;
+		}
+		final Timestamp minLoadingDateFromCompletedDeliveryInstructions = queryBL.createQueryBuilder(I_M_Delivery_Planning.class)
```

Comment:
i think we can refactor the minLoadingDateFromCompletedDeliveryInstructions calculation to a separate method which can sit in this Repository because i assume I_M_Delivery_Planning and I_M_ShipperTransportation are managed by this Repository.

Reply:
done.